### PR TITLE
test(autoware_motion_velocity_planner_common): cover utils.cpp pure logic and CollisionChecker

### PR DESCRIPTION
**Pull request by youtalk**
## Description

Part of the autoware_core quality campaign (testability + coverage), one ROS package per PR.

Adds a focused gtest suite for the previously-untested pure logic in `utils.cpp` — `calc_object_possible_max_dist_from_center` (incl. its throw), `get_index_with_longitudinal_offset` edge cases, `get_extended_trajectory_points`, `calc_distance_to_front_object`, and `concat_vectors` — plus enabled correctness tests for `CollisionChecker::get_collisions` (point / line / empty-rtree / no-collision), replacing the previously DISABLED-only coverage.

Extracts the string→`ObjectClassification` mapping of `get_target_object_type` into a new pure free function `classification_types_from_flags(std::map<std::string,bool>)` so the mapping is unit-testable without a live node. `get_target_object_type` keeps its existing public signature and still declares the same parameters, now delegating to the pure helper.

## Related links

**Parent Issue:** autowarefoundation/autoware_core#1096

## How was this PR tested?

`colcon build` + `colcon test --packages-select autoware_motion_velocity_planner_common` in the `autoware:core-devel-jazzy` container — new and re-enabled tests pass.

## Notes for reviewers

The only observable change is described below; everything else is additive tests + a pure-function extraction.

## Interface changes

None (public signature of `get_target_object_type` unchanged; `classification_types_from_flags` is additive).

## Effects on system behavior

As a side effect of the extraction, the classification list returned by `get_target_object_type` is now produced in a fixed canonical order instead of depending on `unordered_map` iteration order. **The set of returned values is unchanged** — only their ordering is now deterministic.

### CODE REVIEW

File: planning/motion_velocity_planner/autoware_motion_velocity_planner_common/test/test_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+TEST(MvpUtilsLongitudinalOffset, BackwardFromDefaultStart)
+{
+  const auto points = make_straight_forward_trajectory(5, 1.0);
+
+  // Negative offset and no start_idx -> start from the last index (4) and walk backward.
+  // offset -1.4: accumulate from index 4: at i=4 sum=1.0 (<1.4), at i=3 sum=2.0 (>=1.4) on segment
+  // [3, 4]. back_length = sum + offset = 2.0 - 1.4 = 0.6 (distance from offset to point 3),
+  // front_length = seg - back = 1.0 - 0.6 = 0.4 (distance from offset to point 4). front_length <
+  // back_length -> the offset is closer to point 4, so the segment-start index i = 3 is returned.
```
[the remark this answers is not in the document]
Comment (youtalk) [reply planning/motion_velocity_planner/autoware_motion_velocity_planner_common/test/test_utils.cpp]: Fixed in `73c33d8a` — corrected the segment to [2, 3] and swapped the front/back endpoint labels (front = distance to point 3 = `points[i]`, back = point 2 = `points[i-1]`).
